### PR TITLE
Feat/kfs 974 subscribe to insert statements

### PR DESCRIPTION
**Pull request by YannickPferr**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

This adds the option for the user to subscribe to a running statement that doesn't return any results. After the statement transitions out of the PENDING state, we wait for a terminal state or until the user detaches by pressing Enter. A statement is considered to be in terminal state in these cases:
- statement status is COMPLETED
- statement status is FAILED
- statement status is RUNNING and next page token is not empty (indicating that it returns results)

While a user is attached to a statement, we print the status detail returned from the backend (in case it's not empty).

### CODE REVIEW

File: internal/pkg/flink/internal/controller/statement_controller.go [hunk 1]
```diff
@@ -100,3 +103,40 @@ func (c *StatementController) isCancelEvent() bool {
...
+}
+
+func (c *StatementController) isDetachEvent() bool {
+	if b, err := c.consoleParser.Read(); err == nil && len(b) > 0 {
+		pressedKey := prompt.Key(b[0])
+
+		switch pressedKey {
+		case prompt.ControlM:
+			fallthrough
+		case prompt.Enter:
```
Comment (brianstrauch) [inline internal/pkg/flink/internal/controller/statement_controller.go:137]: Not sure we need a `switch` if there's only one branch.
```suggestion
		case prompt.ControlM, prompt.Enter:
```

File: internal/pkg/flink/internal/store/store.go [hunk 2]
```diff
@@ -294,3 +290,44 @@ func (s *Store) propsDefault(propsWithoutDefault map[string]string) map[string]s
...
+
+			if len(capturedErrors) > capturedErrorsLimit {
+				return nil, &types.StatementError{
+					Message: fmt.Sprintf("the server can't process this statement right now, exiting after %d retries",
+						len(capturedErrors)),
+					FailureMessage: fmt.Sprintf("captured retryable errors: %s", strings.Join(capturedErrors, "; ")),
+				}
+			}
+
+			time.Sleep(1 * time.Second)
```
Comment (brianstrauch) [inline internal/pkg/flink/internal/store/store.go:328]:
```suggestion
			time.Sleep(time.Second)
```

**Review by swist — APPROVED** [review]

File: internal/pkg/flink/internal/controller/.snapshots/TestStatementControllerTestSuite-TestExecuteStatementReturnsWhenUserDetaches [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Statement successfully submitted.
+Waiting for statement to be ready. Statement phase is PENDING.
+Statement phase is RUNNING. Press return to detach.
```
[the remark this answers is not in the document]
Comment (YannickPferr) [reply internal/pkg/flink/internal/controller/.snapshots/TestStatementControllerTestSuite-TestExecuteStatementReturnsWhenUserDetaches:3]: Sounds good to me, will change!

**Comment by Guttz** [comment]
Let's print the error as long as the user doesn't detach instead of only 5 times. Also, maybe add a hint that we're listening for errors.

File: internal/pkg/flink/internal/controller/.snapshots/TestStatementControllerTestSuite-TestExecuteStatementReturnsWhenUserDetaches (hunk 3 above)
[the remark this answers is not in the document]
Comment (YannickPferr) [reply internal/pkg/flink/internal/controller/.snapshots/TestStatementControllerTestSuite-TestExecuteStatementReturnsWhenUserDetaches]: Done!

**Comment by YannickPferr** [comment]
> Didn't have the time to test it on stag but code looks good

No worries, you can test it next week! No need to rush this out on a Friday evening and risk breaking something 😅

**Pull request merged**